game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -44,15 +46,22 @@ public void setDiceRoll(final DiceRoll diceRoll) {
     SwingAction.invokeNowOrLater(() -> {
       removeAll();
       for (int i = 1; i <= data.getDiceSides(); i++) {
-        final List<Die> dice = diceRoll.getRolls(i);
+        final int index = i;
+        final List<Die> dice = Optional.ofNullable(diceRoll)
+            .map(roll -> roll.getRolls(index))
+            .orElse(Collections.emptyList());
         if (dice.isEmpty()) {
           continue;
         }
         add(new JLabel("Rolled at " + (i) + ":"));
         add(create(diceRoll.getRolls(i)));
```

Comment:
Instead of checking all rolls again we should pass 'dice' here

Reply:
You're right, but this branch is a NPE bug fix, I had not considered that refactor at all. I was keeping the scope pretty contained to just that. While here this can be fixed, but opening up pandoras box really for further improvements, I'm pretty sure we could nitpick this down to just a few lines that would make sense, but it would be risky and well beyond the scope of the NPE fix.

Reply:
`ifPresent` helps this even more, pushed: ff88eff